sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/v2/V2SessionCatalog.scala

```diff
@@ -153,6 +152,18 @@ class V2SessionCatalog(sessionState: SessionState) extends TableCatalog {
     }
   }
 
+  override def renameTable(oldIdent: Identifier, newIdent: Identifier): Unit = {
+    if (loadTable(newIdent) != null) {
+      throw new TableAlreadyExistsException(newIdent)
+    }
+
+    if (loadTable(oldIdent) != null) {
```

Comment:
why aren't you using the same pattern of other cases, ie `Optional ... match`?

Reply:
Changed to the `Optional ... match pattern`.